node/actors/network/src/frame.rs

```diff
@@ -53,28 +54,41 @@ pub(crate) async fn recv_proto<T: zksync_protobuf::ProtoFmt, S: io::AsyncRead +
     ctx: &ctx::Ctx,
     stream: &mut S,
     max_size: usize,
-) -> anyhow::Result<T> {
+) -> ctx::Result<T> {
     let mut msg_size = [0u8; 4];
-    io::read_exact(ctx, stream, &mut msg_size).await??;
+    io::read_exact(ctx, stream, &mut msg_size)
+        .await?
+        .context("read_exact(len)")?;
     let msg_size = u32::from_le_bytes(msg_size);
-    anyhow::ensure!(
-        msg_size as usize <= max_size,
-        "message too large: max = {max_size}, got {msg_size}",
-    );
+    if msg_size as usize > max_size {
+        return Err(
+            anyhow::format_err!("message too large: max = {max_size}, got {msg_size}",).into(),
+        );
+    }
```

Comment:
What was wrong with `anyhow::ensure!`?

Reply:
anyhow::ensure needs the function to return exactly anyhow::Result (with ctx::Result it won't compile). It is possible to write a more general macro which will work with any error type implementing `From<anyhow::Error>`.